packages/vats/src/lib-chainStorage.js

```diff
@@ -0,0 +1,70 @@
+// @ts-check
+
+import { Far } from '@endo/far';
+
+const { details: X } = assert;
+
+// TODO: Formalize segment constraints.
+// Must be nonempty and disallow (unescaped) `.`, and for simplicity
+// (and future possibility of e.g. escaping) we currently limit to
+// ASCII alphanumeric plus underscore.
+const pathSegmentPattern = /^[a-zA-Z0-9_-]{1,100}$/;
+
+/**
+ * Create a root storage node for a given backing function and root key.
+ *
+ * @param {(message: any) => any} toStorage a function for sending a storageMessage object to the storage implementation (cf. golang/cosmos/x/swingset/storage.go)
+ * @param {string} rootKey
+ */
+export function makeChainStorageRoot(toStorage, rootKey) {
+  assert.typeof(rootKey, 'string');
+
+  function makeChainStorageNode(key) {
+    const node = {
+      getKey() {
+        return key;
+      },
```

Comment:
```suggestion
      getStoreKey() {
        // This duplicates the Go code at
        // https://github.com/Agoric/agoric-sdk/blob/cb272ae97a042ceefd3af93b1b4601ca49dfe3a7/golang/cosmos/x/swingset/keeper/keeper.go#L295
        return { storeName: "swingset", storeSubkey: `swingset/data:${path}` } ;
      },
```

Reply:
is that colon a *third* separator character, visible in the RPC queries?